pkg/broker/handler/pool/pool_test.go

```diff
@@ -55,6 +56,7 @@ func TestSyncPool(t *testing.T) {
 			t.Errorf("SyncOnce was called more than once")
 		}
 		ch <- struct{}{}
+		time.Sleep(500 * time.Millisecond)
```

Comment:
I haven't looked into the root cause of this yet. Is it easy to make the test 100% deterministic?

Reply:
After receiving a signal, the fake pool will increase its count asynchronously. And we were checking right away previously, which caused the flakiness. I don't see an obvious way to make it deterministic other than longer waiting.

Reply:
One way to do this is to have a channel in the fakePool, and send a signal to the channel each time `SyncOnce` is called. The test can wait for the signals in the channel.

Normally unit test should be deterministic. Otherwise it may bite us again.

Reply:
I agree in principal. But in this case it's also possible our code has a bug and `SyncOnce` was never called and we would wait infinitely. To help we have to add a timeout, which is similar to what it's doing now. That's why I think it's not necessary to introduce such a mechanism.

Reply:
One solution could be to use a buffered channel instead of an atomic counter in fakeSyncPool so that the test could block waiting for each sync instead of sleeping. Yes ideally this would also introduce a test timeout but I think this would be preferable to sleeping.

Reply:
Fine with me. We can implement that if we see this again.

Reply:
I made the change you suggested to save some time